src/pages/Compare/index.spec.tsx

```diff
@@ -886,6 +916,191 @@ describe(__filename, () => {
   });
 
+  describe('preload', () => {
+    const setUpForPreloadAndRender = ({
+      addonId = 2,
+      baseVersionId = 4,
+      buildTree = true,
+      currentFile = 'currentFile.json',
+      headVersionId = 5,
+      loadDiff = true,
+      loadVersionFile = true,
+      nextModifiedFile = 'modifiedFile.json',
+      ...params
+    }) => {
+      const version = createExternalVersionWithEntries(
+        [
+          { path: currentFile, status: 'M' },
+          { path: 'fileWithoutChange', status: '' },
+          { path: nextModifiedFile, status: 'M' },
+        ],
+        { id: headVersionId, selected_file: currentFile },
+      );
+      const {
+        _fetchDiff,
+        _fetchVersionFile,
+        dispatchSpy,
+        fakeThunk,
+      } = loadDiffAndRender({
+        addonId,
+        baseVersionId,
+        buildTree,
+        headVersionId,
+        loadDiff,
+        loadVersionFile,
+        version,
+        ...params,
+      });
+      return {
+        _fetchDiff,
+        _fetchVersionFile,
+        dispatchSpy,
+        fakeThunk,
+      };
+    };
+
+    it('dispatches fetchDiff and fetchVersionFile for the next file with diff', () => {
+      const addonId = 10;
+      const baseVersionId = 44;
+      const headVersionId = 55;
+      const currentFile = 'current.json';
+      const nextModifiedFile = 'modified.json';
+      const {
+        _fetchDiff,
+        _fetchVersionFile,
+        dispatchSpy,
+        fakeThunk,
+      } = setUpForPreloadAndRender({
+        addonId,
+        baseVersionId,
+        currentFile,
+        headVersionId,
+        nextModifiedFile,
+      });
+
+      expect(dispatchSpy).toHaveBeenCalledWith(fakeThunk.thunk);
+      expect(_fetchDiff).toHaveBeenCalledWith({
+        addonId,
+        baseVersionId,
+        headVersionId,
+        path: nextModifiedFile,
+      });
+      expect(_fetchVersionFile).toHaveBeenCalledWith({
+        addonId,
+        versionId: headVersionId,
+        path: nextModifiedFile,
+      });
+    });
+
+    it.each([['buildTree'], ['loadVersionFile'], ['loadDiff']])(
+      'does not dispatch fetchDiff or fetchVersionFile for the next file before %s is dispatched',
+      (action) => {
+        const currentFile = 'current.json';
+        const nextModifiedFile = 'next.json';
+        const { _fetchDiff, _fetchVersionFile } = setUpForPreloadAndRender({
+          currentFile,
+          nextModifiedFile,
+          [action]: false,
+        });
+
+        expect(_fetchDiff).not.toHaveBeenCalledWith(
+          expect.objectContaining({
+            path: nextModifiedFile,
+          }),
+        );
+        expect(_fetchVersionFile).not.toHaveBeenCalledWith(
+          expect.objectContaining({
+            path: nextModifiedFile,
+          }),
+        );
+      },
+    );
+
+    it('does not dispatch fetchVersionFile for the next file if it is loading', () => {
+      const headVersionId = 2222;
+      const baseVersionId = 11;
```

Comment:
Is `baseVersionId` needed for the test? I don't see it in any assertions.

Reply:
I keep `baseVersionId` in this test to make sure `headVersionId`>`baseVersionId`. Otherwise, [`loadData()` returns earlier](https://github.com/mozilla/addons-code-manager/blob/f5f1ffe25415311439a76c0ffcf3bcd61a2d5401/src/pages/Compare/index.tsx#L114), and it will not reach[ the conditional state we would like to test](https://github.com/mozilla/addons-code-manager/blob/f5f1ffe25415311439a76c0ffcf3bcd61a2d5401/src/pages/Compare/index.tsx#L160). Do you think it is necessary?

Reply:
Ah, yes, I see now. Thanks for the links.